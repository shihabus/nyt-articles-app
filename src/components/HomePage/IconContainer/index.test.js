/* eslint-disable no-undef */
// /* eslint-disable import/named */
// /* eslint-disable no-undef */
// /**
//  * @jest-environment jsdom
//  */

import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Icon from "./index";
import { findComponent } from "../../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (props = {}) => {
  const wrapper = shallow(<Icon {...props} />);
  return wrapper;
};

describe("Icon", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot("Icon");
  });

  it('should render child',()=>{
    expect(findComponent(wrapper,"Icon").length).toEqual(1)
  })
});
