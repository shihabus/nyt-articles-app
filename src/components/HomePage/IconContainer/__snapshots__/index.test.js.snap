// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Snapshot: Icon 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "10%",
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="gray"
    name="chevron-right"
    size={15}
    style={
      Object {
        "alignSelf": "center",
      }
    }
    testID="Icon"
  />
</View>
`;
